source/conf.ts

```diff
@@ -0,0 +1,531 @@
+		if (typeof callback !== 'function') {
+			throw new TypeError(`Expected \`callback\` to be of type \`function\`, got ${typeof callback}`);
+		}
+
+		return this._handleChange(() => this.store, callback);
+	}
+
+	get size(): number {
+		return Object.keys(this.store).length;
+	}
+
+	get store(): T {
+		try {
+			const data = fs.readFileSync(this.path, this._encryptionKey ? null : 'utf8');
+			const dataString = this._encryptData(data);
+			const deserializedData = this._deserialize(dataString);
+			this._validate(deserializedData);
+			return Object.assign(createPlainObject(), deserializedData);
+		} catch (error) {
+			if (error.code === 'ENOENT') {
+				this._ensureDirectory();
+				return createPlainObject();
+			}
+
+			if (this._options.clearInvalidConfig && error.name === 'SyntaxError') {
+				return createPlainObject();
+			}
+
+			throw error;
+		}
+	}
+
+	set store(value: T) {
+		this._ensureDirectory();
+
+		this._validate(value);
+		this._write(value);
+
+		this._events.emit('change');
+	}
+
+	* [Symbol.iterator](): Iterator<[keyof T, T[keyof T]]> {
+		for (const [key, value] of Object.entries(this.store)) {
+			yield [key, value];
+		}
+	}
+
+	private _encryptData(data: string | Buffer): string {
+		if (!this._encryptionKey) {
+			return data.toString();
+		}
+
+		try {
+			// Check if an initialization vector has been used to encrypt the data
+			if (this._encryptionKey) {
+				try {
+					if (data.slice(16, 17).toString() === ':') {
+						const initializationVector = data.slice(0, 16);
+						const password = crypto.pbkdf2Sync(this._encryptionKey, initializationVector.toString(), 10000, 32, 'sha512');
+						const decipher = crypto.createDecipheriv(encryptionAlgorithm, password, initializationVector);
+						data = Buffer.concat([decipher.update(Buffer.from(data.slice(17))), decipher.final()]).toString('utf8');
+					} else {
+						const decipher = crypto.createDecipher(encryptionAlgorithm, this._encryptionKey);
+						data = Buffer.concat([decipher.update(Buffer.from(data)), decipher.final()]).toString('utf8');
+					}
+				} catch (_) { }
+			}
+		} catch {
+		}
+
+		return data.toString();
+	}
+
+	private _handleChange<TKey extends keyof T>(
+		getter: () => T | T[TKey] | undefined,
+		callback: OnDidChangeCallback<any>
+	): () => EventEmitter {
+		let currentValue = getter();
+
+		const onChange = (): void => {
+			const oldValue = currentValue;
+			const newValue = getter();
+
+			try {
+				// TODO: Use `util.isDeepStrictEqual` when targeting Node.js 10
+				assert.deepEqual(newValue, oldValue);
+			} catch (_) {
+				currentValue = newValue;
+				callback.call(this, newValue, oldValue);
+			}
+		};
+
+		this._events.on('change', onChange);
+		return () => this._events.removeListener('change', onChange);
+	}
+
+	private readonly _deserialize: Deserialize<T> = value => JSON.parse(value);
+	private readonly _serialize: Serialize<T> = value => JSON.stringify(value, null, '\t');
+
+	private _validate(data: T): void {
```

Comment:
Can you use private class fields instead for these? Now available in TS 3.8. Then we can also drop the `_` prefix.